letsencrypt/models/letsencrypt.py

```diff
@@ -4,165 +4,135 @@
-            )
-            crt.write(chain_cert.read())
-            chain_cert.close()
-            _logger.info('wrote %s', crt.name)
-        reload_cmd = self.env['ir.config_parameter'].get_param(
-            'letsencrypt.reload_command', False)
-        if reload_cmd:
-            _logger.info('reloading webserver...')
-            self.call_cmdline(['sh', '-c', reload_cmd])
+                    'web.base.url', 'localhost')).netloc
+        self.validate_domain(domain)
+        # Generate account key
+        account_key_file = self._generate_key('account_key')
+        account_key = jose.JWKRSA.load(open(account_key_file).read())
+        # Generate domain key
+        domain_key_file = self._generate_key(domain)
+        net = client.ClientNetwork(account_key)
+        acme = client.ClientV2(
+            V2_STAGING_DIRECTORY_URL,
+            net)
+        _logger.info('Auto-accepting TOS')
+        # TODO if you try to register when you already have, this will crash.
+        acme.new_account(NewRegistration())
+        from acme import crypto_util
+        csr = crypto_util.make_csr(open(domain_key_file).read(), [domain])
+        authzr = acme.new_order(csr)
+        # write the file with the challenge
+        challenge_dir = _get_challenge_dir()
+        if not os.path.isdir(challenge_dir):
+            os.makedirs(challenge_dir)
+        # if the domain contains a wildcard, do dns validation
+        if '*' not in domain:
+            challenge = filter(
+                    lambda x: x.chall.typ == 'http-01',
+                    authzr.authorizations[0].body.challenges)[0]
+            from datetime import datetime, timedelta
+            challenge_expiration = datetime.now() + timedelta(minutes=5)
+            token = base64.urlsafe_b64encode(challenge.token)
+            challenge_file = os.path.join(_get_challenge_dir(), '%s' % token)
+            with open(challenge_file, 'wb') as challenge_file:
+                # remove the trailing equals sign (=)
+                challenge_file.write(token + '.' + base64.urlsafe_b64encode(
+                    account_key.thumbprint())[:-1])
         else:
-            _logger.info('no command defined for reloading webserver, please '
-                         'do it manually in order to apply new certificate')
+            challenge_expiration = authzr.authorizations[0].body.expires
+            token = authzr.authorizations[0].body.challenges[0].token
+            # both challenges are DNS challenges, pick the first one
+            token = base64.b64encode(token)
+            # create the dns_record
+            from base_config_settings import BaseConfigSettings
+            company_id = self.env['res.users'].browse(
+                self.env.uid).company_id
+            getattr(BaseConfigSettings,company_id.dns_function)(
+                company_id.dns_login,
+                company_id.dns_key,
+                '.'.join(domain.split('.')[-2:]),
```

Comment:
i doubt it's a good idea to preprocess this - give full information to the function you call, you never know when it's relevant